pkgs/development/libraries/onnxruntime/default.nix

```diff
@@ -30,11 +30,17 @@ let
     sha256 = "sha256-BYL7wxsYRI45l8C3VwxYIIocn5TzJnBtU0UZ9pHwwZw=";
   };
 
-  eigen = fetchFromGitLab {
-    owner = "libeigen";
-    repo = "eigen";
-    rev = "d10b27fe37736d2944630ecd7557cefa95cf87c9";
-    sha256 = "sha256-Lmco0s9gIm9sIw7lCr5Iewye3RmrHEE4HLfyzRkQCm0=";
+  eigen' = applyPatches {
+    name = "patch-build-error.patch";
+    src = eigen.src;
+    patches = eigen.patches or [] ++ [
+      # Patch for Eigen seems to be necessary to avoid build errors as referenced here:
+      # https://aur.archlinux.org/pkgbase/python-onnxruntime?O=30#comment-856116
+      (fetchpatch {
+        url = "https://gitlab.com/libeigen/eigen/-/commit/d0e3791b1a0e2db9edd5f1d1befdb2ac5a40efe0.patch";
+        sha256 = "sha256-8qiNpuYehnoiGiqy0c3Mcb45pwrmc6W4rzCxoLDSvj0=";
+      })
```

Comment:
Can we patch `eigen` instead of creating an override here?

Reply:
Hm, apparently not, since we need a patched source. But patching only happens during build, so a patched `eigen` will not provide a patched `eigen.src`.